doc/concepts/pipeline-concepts/datum/relationship-between-datums.md

```diff
@@ -0,0 +1,159 @@
+overwrite that file but appends it to the file `foo` in the
+repo. Therefore, the size of the file `foo` doubles and
+becomes 200 KB. Pachyderm enables you to overwrite files as
+well by using the `--overwrite` flag. The order of processing
+is not guaranteed, and all datums are processed randomly.
+For more information, see [File](file.html).
+
+When new data comes in, a Pachyderm pipeline automatically
+starts a new job. Each Pachyderm job consists of the
+following stages:
+
+1. Creation of input datums. In this stage, Pachyderm breaks
+input files into datums according to the glob pattern setting
+in the pipeline specification.
+1. Transformation. The pipeline uses your code to processes the
+datums.
+1. Creation of output datums. Pachyderm creates file or files from the
+processed data and combines them into output datums.
+1. Merge. Pachyderm combines all files with the same file path
+by appending, overwriting, or deleting them to create the final commit.
+
+If you think about this process in terms of filesystem objects and
+processing abstractions, the following transformation happens:
+
+**input files = > input datums => output datums => output files**
+
+This section provides examples that help you understand such fundamental
+Pachyderm concepts as the datum, incremental processing, and phases of
+data processing.
+
+## Example 1: One file in the input datum, one file in the output datum
+
+The simplest example of datum processing is when you have one file in
+the input datum that results in one file in the output datum.
+In the diagram below, you can see three input datums, each of which
+includes one file, that result in three output datums. Whether you have
+submitted all these datums in a single or multiple commits, the final
+result remains the same—three datums, each of which has one file.
+
+In the diagram below, you can see the following datums:
+
+ - `datum 1` has one file and results in one file in one output datum.
+ - `datum 2` has one file and results in one file in one output datum.
+ - `datum 3` has one file and results in one file in one output datum.
+
+![One to one](../../../images/d_datum_processing_one_to_one.svg)
+
+If you decide to overwrite a single line in the file in `datum 3` and
+add `datum 4`, Pachyderm sees the four datums and checks them for changes
+one-by-one. Pachyderm verifies that there are no changes in `datum 1` and
+`datum 2` and skips these datums. Pachyderm detects changes in the
+`datum 3` and the `--overwrite` flag and replaces the `datum 3` with the
+new `datum 3'`. When it detects `datum 4` as a completely new datum,
+it processes the whole datum as new. Although only two datums were
+processed, the output commit of this change contains all four files.
+
+![One to one overwrite](../../../images/d_datum_processing_one_to_one_overwrite.svg)
+
+## Example 2: One file in the input datum, multiple files in the output datum
+
+Some pipelines ingest one file in one input datum and create multiple
```

Comment:
Making a bunch of tweaks to match the other section and comments. Everything you had here was accurate and correct, I just reworded a few things to try have it flow a bit smoother. There is definitely a lot of craziness going on here and it's pretty hard to wrangle. :) We might want to take another pass together to tighten it up even further. Also note, we should note files as `file 1` just like we do `datum 1`. If I missed that anywhere else in this doc, please correct it there.

---------

Often times, a job will purposely want different output datums to write to the same file path and then merge those results togethers. The merge step may include appending certain files together or overwriting outdated chunks of files.

In the diagram below, we again have 3 input datums, but in this case each output datum includes 2 files with overlapping names. 

 - `datum 1` results in `file 1` and `file 3`.
- `datum 2` results in `file 2` and `file 3`.
- `datum 3` results in `file 2` and `file 1`.

Note that the content of `file 1` in `datum 1` will likely not be the same content as `file 1` in `datum 3` since these datums used different input data. Pachyderm will then need to merge (in this append) the content of these two files together to create a final output `file 1` that you'll see in the output commit. Similar merges will need to happen for `file 2` and `file 3`. 

 ![One to many](../../../images/d_datum_processing_one_to_many.svg)


If in your next input commit, let's say you decide to make a change to some of the input files in `datum 2`.  Because `datum 1` and `datum 3` are
unchanged, Pachyderm skips processing these datums, but it detects that `datum 2` is changed and processes it. Based on the new input data, let's say that the output of `datum 2` now includes three files, `file 1`, file `3` and `file 4`. Pachyderm then needs to re-merge all these files together to create the new final output files. A few key notes about this re-merge:

- `file 1` now has three chunks, the previous two from `datum 1` and `datum 3` which have not changed, but also the third new chunk from `datum 2'`. 
- The `file 2` chunk from `datum 2` is deleted leaving only the `file 2` from `datum 3`. 
- `file 3` still has two chunks, but the new `file 3` chunk from `datum 2'` replaces the previous one.
- `file 4` is new and only has one chunk.

 ![One to many](../../../images/d_datum_processing_one_to_many_overwrite.svg)